umpost/conversion_driver_esm1p5.py

```diff
@@ -117,14 +130,73 @@ def find_matching_fields_files(dir_contents, fields_file_name_pattern):
     return fields_file_paths
 
 
-def convert_fields_file_list(fields_file_paths, nc_write_dir):
+def get_nc_filename(fields_file_name, date=None):
     """
-    Convert group of fields files to NetCDF, writing output in nc_write_dir.
+    Format a netCDF output filename based on the input fields file name and
+    its date. Assumes fields_file_name follows ESM1.5's naming convention
+    '{5 char run_id}.pa{unit}{date encoding}`.
 
     Parameters
     ----------
-    fields_file_paths : list of paths to fields files for conversion.
-    nc_write_dir : directory to save NetCDF files into.
+    fields_file_name: name of fields file to be converted.
+    date: tuple of form (year, month, day) associated with fields file data.
+
+    Returns
+    -------
+    name: formated netCDF filename for writing output.
+    """
+    if date is None:
+        # If no date information supplied, just append ".nc"
+        # to current file name.
+        return f"{fields_file_name}.nc"
+
```

Comment:
> Tweak get_nc_write_path(fields_file_path, nc_write_dir) to get_nc_write_path(fields_file_path, nc_write_dir, date=None)
If the date input is None, default to the prior behaviour of adding .nc

I've added the naming logic into the `get_nc_filename` setting, so that the new naming rules are applied when a date is supplied, while it reverts to the old behaviour if a date is not supplied.

@truth-quark do you think it's worth supporting both of these cases? From I've seen, output fields files from ESM1.5 will always contain a date.

Reply:
After trialing removing the `date = None` option in [this branch](https://github.com/ACCESS-NRI/um2nc-standalone/blob/afdb0e22e8f6c251b0e6a23bfaa3adda48b76971/umpost/conversion_driver_esm1p5.py#L126-L161) I'm wondering whether it's simplest to actually keep the `date=None` option in?

> I'd wrap `year, month, _ = date` in a `try/except AttributeError` to catch anything passed in without a date interface.

The main part I'm unsure about is the best way for the exceptions to be handled. E.g. if `date=None` was supplied at this point, I think we'd get a `TypeError`, but in the current code I'm not sure if there's a clear way for this to happen. Let me know if the experiment in [this branch](https://github.com/ACCESS-NRI/um2nc-standalone/blob/afdb0e22e8f6c251b0e6a23bfaa3adda48b76971/umpost/conversion_driver_esm1p5.py#L126-L161) looks reasonable/similar to what you had in mind though!

Reply:
> After trialing removing the `date = None` option in [this branch](https://github.com/ACCESS-NRI/um2nc-standalone/blob/afdb0e22e8f6c251b0e6a23bfaa3adda48b76971/umpost/conversion_driver_esm1p5.py#L126-L161) I'm wondering whether it's simplest to actually keep the `date=None` option in?

Leaving `date=None` in is fine as the func provides a sensible default result. Let's leave that part in.

Reply:
> > I'd wrap `year, month, _ = date` in a `try/except AttributeError` to catch anything passed in without a date interface.
> 
> The main part I'm unsure about is the best way for the exceptions to be handled. E.g. if `date=None` was supplied at this point, I think we'd get a `TypeError`, but in the current code I'm not sure if there's a clear way for this to happen. Let me know if the experiment in [this branch](https://github.com/ACCESS-NRI/um2nc-standalone/blob/afdb0e22e8f6c251b0e6a23bfaa3adda48b76971/umpost/conversion_driver_esm1p5.py#L126-L161) looks reasonable/similar to what you had in mind though!

Ah yes, `TypeError` occurs when attempting to unpack a non-iter `None`. Trying `None.foo()` in a REPL raises an `AttributeError` as `None` is an instance of `NoneType`. Both cases can be captured in a multi error type handler:

```Python
try:
    # unpack None or possible non date obj access
except (TypeError, AttributeError)  as err:
    # some kind of handling
```

I'd also include the date value in the error message for debugging assistance.

Also, you could modify the parameter docs to indicate `None` & `date` are the expected args. This leaves users to debug passing in the wrong thing :-)

Reply:
> Also, you could modify the parameter docs to indicate None & date are the expected args

I've updated the docstrings to be a bit more clear about `None` being allowed in e9d0d79. 

I just wanted to check whether I've understood things correctly – since we're supporting the `date=None` case in the following, can we avoid adding the extra error handling?

https://github.com/ACCESS-NRI/um2nc-standalone/blob/e9d0d79627f0a6097991cdf687329d2311a30663/umpost/conversion_driver_esm1p5.py#L142-L143

Reply:
> I just wanted to check whether I've understood things correctly – since we're supporting the `date=None` case in the following, can we avoid adding the extra error handling?

Oooops, a mental shortcut meant wrongly assuming a `datetime.date` obj was being passed in & thinking about a wrong solution <facepalm>.

Since the date passing is handled internally, let's assume the those functions are internal & not used as an API. Thus, there's no requirement to do the extra Type/AttributeError checks.